### PR TITLE
Fix the use of -mtowncrier. #163

### PR DESCRIPTION
**Pull request by nedbat**
This makes the -m invocation work.

I added a test, but:
- I didn't know in what file you wanted it
- I didn't know if Trial had a way to manage the current directory

Let me know if I should change anything.

### CODE REVIEW

**Comment by charris** [comment]
Also need this.

**Comment by nedbat** [comment]
@charris I don't know what you are referring to.  Is there a larger conversation somewhere? Or did you mean to link to something that should be included in this PR?

**Comment by charris** [comment]
I run several versions of Python for testing and development, plus upgrading the OS twice a year. In that circumstance it is helpful to have towncrier work as a module so as to keep track of which Python it is installed with.

File: src/towncrier/test/test_project.py [hunk 1]
```diff
@@ -36,3 +37,24 @@ def test_tuple(self):
...
+        temp = self.mktemp()
+        new_dir = os.path.join(temp, "dashm")
+        os.makedirs(new_dir)
+        orig_dir = os.getcwd()
+        try:
+            os.chdir(new_dir)
+            with open("pyproject.toml", "w") as f:
+                f.write('[tool.towncrier]\n' 'directory = "news"\n')
+            os.makedirs("news")
+            out = check_output(["python", "-m", "towncrier", "--help"])
```
Comment (cdunklau) [inline src/towncrier/test/test_project.py]: Probably should use `sys.executable` instead of `'python'` here, to avoid relying on PATH

Comment (cdunklau) [reply src/towncrier/test/test_project.py]: ...hmm I didn't know this would show up as a "review", my intention was just to comment.

Comment (glyph) [inline src/towncrier/test/test_project.py:56]:
```suggestion
            out = check_output([sys.executable, "-m", "towncrier", "--help"])
```

File: src/towncrier/test/test_project.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 # See LICENSE for details.
 
 import os
```
Comment (glyph) [inline src/towncrier/test/test_project.py:4]:
```suggestion
import os
import sys
```

File: src/towncrier/test/test_project.py [hunk 3]
```diff
@@ -36,3 +37,24 @@ def test_tuple(self):
 
         version = get_version(temp, "mytestproja")
         self.assertEqual(version, "1.3.12")
+
+
+class InvocationTests(TestCase):
+    def test_dash_m(self):
+        """
+        `python -m towncrier` works.
```
Comment (glyph) [inline src/towncrier/test/test_project.py:45]: "works" is one of those non-words called out by https://jml.io/pages/test-docstrings.html so let's say what working means

```suggestion
        `python -m towncrier` invokes the main entrypoint.
```

**Pull request merged**